cmd/disk-cache-backend.go

```diff
@@ -595,11 +734,20 @@ func (c *diskCache) bitrotReadFromCache(ctx context.Context, filePath string, of
 func (c *diskCache) Get(ctx context.Context, bucket, object string, rs *HTTPRangeSpec, h http.Header, opts ObjectOptions) (gr *GetObjectReader, err error) {
 	var objInfo ObjectInfo
 	cacheObjPath := getCacheSHADir(c.dir, bucket, object)
+	var rngInfo rangeInfo
 
-	if objInfo, err = c.Stat(ctx, bucket, object); err != nil {
+	if objInfo, rngInfo, err = c.statRange(ctx, bucket, object, rs); err != nil {
 		return nil, toObjectErr(err, bucket, object)
 	}
-
+	cacheFile := cacheDataFile
+	objSize := objInfo.Size
+	if (rngInfo != rangeInfo{}) {
```

Comment:
we should avoid doing struct comparisons like this as these are buggy and have side effects. Can you simply do `rngInfo.Empty()` ?